### PR TITLE
Update docs and add `shutdown_duration` field

### PR DESCRIPTION
**Pull request by Uberi**
As per the email sent this morning, this PR adds the shutdown_duration field, and also documents the ping field mapping.

Ideally we'd want some more precise documentation than "here's where to look in the code", but this should be enough to get started with.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,3 +40,20 @@ If you run into memory issues during compilation time issue the following comman
...
 ```
+
+The Longitudinal Derived Stream
+-------------------------------
+
+The longitudinal dataset is generated by `src/main/scala/streams/Longitudinal.scala`, driven by `src/main/scala/DerivedStream.scala`. It's useful for running fast queries on common fields in pings, such as 
+
+The resulting dataset can be retrieved as a [DataFrame](http://pandas.pydata.org/pandas-docs/stable/dsintro.html#dataframe), which can be thought of as an array of map of arrays:
+
+* The outer array (which is actually a DataFrame) contains one element per client (clients are distinguished by the `clientId` top-level field in pings).
```
Comment (vitillo) [inline README.md]: It's a Spark DataFrame, not a pandas one.

File: README.md [hunk 2]
```diff
@@ -40,3 +40,20 @@ If you run into memory issues during compilation time issue the following comman
 ```
 export JAVA_OPTIONS="-Xss128M -Xmx2048M" 
 ```
+
+The Longitudinal Derived Stream
+-------------------------------
+
+The longitudinal dataset is generated by `src/main/scala/streams/Longitudinal.scala`, driven by `src/main/scala/DerivedStream.scala`. It's useful for running fast queries on common fields in pings, such as 
+
+The resulting dataset can be retrieved as a [DataFrame](http://pandas.pydata.org/pandas-docs/stable/dsintro.html#dataframe), which can be thought of as an array of map of arrays:
```
Comment (vitillo) [inline README.md]: It's a Spark DataFrame, not a pandas one.

File: README.md [hunk 3]
```diff
@@ -40,3 +40,20 @@ If you run into memory issues during compilation time issue the following comman
 ```
 export JAVA_OPTIONS="-Xss128M -Xmx2048M" 
 ```
+
+The Longitudinal Derived Stream
+-------------------------------
+
+The longitudinal dataset is generated by `src/main/scala/streams/Longitudinal.scala`, driven by `src/main/scala/DerivedStream.scala`. It's useful for running fast queries on common fields in pings, such as
```
Comment (vitillo) [inline README.md]: such as... ?

File: README.md [hunk 4]
```diff
@@ -40,3 +40,20 @@ If you run into memory issues during compilation time issue the following comman
...
+
+* The outer array (which is actually a DataFrame) contains one element per client (clients are distinguished by the `clientId` top-level field in pings).
+* Each element of that outer array is a map, where fields are the ones defined in `src/main/scala/streams/Longitudinal.scala` in the `buildSchema` function. Remember that the map represents a single client.
+  * Whenever records are converted from JSON, JSON keys are converted from camelCase (in the original ping) into snake_case (in the resulting dataset) by `src/main/scala/avro/JSON2Avro.scala`.
+  * For example, the `firstPaint` simple measure in a ping object like `(payload.simpleMeasurements -> {"firstPaint":2114,...other simple measurements...},...other fields..)` shows up under `first_paint` in the resulting dataset.
+  * The `buildRecord` function in `src/main/scala/streams/Longitudinal.scala` defines exactly how fields in the pings map to fields in the schema.
+* Each value in the map is itself an array, where elements are the values of that field in the map for every ping sent by that client.
```
Comment (vitillo) [inline README.md]: Most values in the map are arrays (e.g. `client_id` is not)

**Comment by vitillo** [comment]
Could you move the README for the longitudinal dataset in the `docs` folder?

**Comment by Uberi** [comment]
Changed:

* Feedback from review.
* Add a note about fields that aren't arrays.

**Comment by Uberi** [comment]
Changed:

* Fix broken link in README.
* Move info about longitudinal dataset to its own folder.

File: docs/Longitudinal.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+The Longitudinal Derived Stream
+-------------------------------
+
+The longitudinal dataset is generated by `src/main/scala/streams/Longitudinal.scala`, driven by `src/main/scala/DerivedStream.scala`. It's useful for running fast queries on common fields in pings, such as user OSs, channels, and settings.
+
+The resulting dataset can be retrieved as a [DataFrame](https://spark.apache.org/docs/1.3.1/api/java/org/apache/spark/sql/DataFrame.html), which can be thought of as a distributed array of map of arrays:
```
Comment (vitillo) [inline docs/Longitudinal.md]: Could you link to the 1.6 version of DataFrame?

**Comment by Uberi** [comment]
Changes:

* Merge master.
* Update spark version.

File: build.sbt [hunk 6]
```diff
@@ -25,6 +25,7 @@ lazy val root = (project in file(".")).
     libraryDependencies += "com.github.nscala-time" %% "nscala-time" % "2.10.0",
     libraryDependencies += "org.rogach" %% "scallop" % "1.0.0",
     libraryDependencies += "com.twitter" %% "algebird-core" % "0.12.0"
+
```
Comment (vitillo) [inline build.sbt]: why did you add a space here?

Comment (Uberi) [inline build.sbt]: Oops, that seems to have snuck in during the merge, I've removed it now.

**Comment by vitillo** [comment]
r+
